front/lib/api/assistant/configuration.ts

```diff
@@ -1030,3 +1030,148 @@ async function _createAgentDataSourcesConfigData(
+        null;
+      if (agentConfiguration.generationConfigurationId) {
+        newGenerationConfiguration = await AgentGenerationConfiguration.create(
+          {
+            prompt: agentConfiguration.generationConfiguration.prompt,
+            providerId: agentConfiguration.generationConfiguration.providerId,
+            modelId: agentConfiguration.generationConfiguration.modelId,
+            temperature: agentConfiguration.generationConfiguration.temperature,
+          },
+          { transaction: t }
+        );
+      }
+
+      let newRetrievalConfiguration: AgentRetrievalConfiguration | null = null;
+      if (agentConfiguration.retrievalConfigurationId) {
+        newRetrievalConfiguration = await AgentRetrievalConfiguration.create(
+          {
+            sId: generateModelSId(),
+            query: agentConfiguration.retrievalConfiguration.query,
+            queryTemplate:
+              agentConfiguration.retrievalConfiguration.queryTemplate,
+            relativeTimeFrame:
+              agentConfiguration.retrievalConfiguration.relativeTimeFrame,
+            relativeTimeFrameDuration:
+              agentConfiguration.retrievalConfiguration
+                .relativeTimeFrameDuration,
+            relativeTimeFrameUnit:
+              agentConfiguration.retrievalConfiguration.relativeTimeFrameUnit,
+            topK: agentConfiguration.retrievalConfiguration.topK,
+            topKMode: agentConfiguration.retrievalConfiguration.topKMode,
+          },
+          { transaction: t }
+        );
+      }
+
+      let newAgentDustAppRunConfiguration: AgentDustAppRunConfiguration | null =
+        null;
+      if (agentConfiguration.dustAppRunConfigurationId) {
+        newAgentDustAppRunConfiguration =
+          await AgentDustAppRunConfiguration.create(
+            {
+              sId: generateModelSId(),
+              appWorkspaceId:
+                agentConfiguration.dustAppRunConfiguration.appWorkspaceId,
+              appId: agentConfiguration.dustAppRunConfiguration.appId,
+            },
+            { transaction: t }
+          );
+      }
+      let newAgentDatabaseQueryConfiguration: AgentDatabaseQueryConfiguration | null =
+        null;
+      if (agentConfiguration.databaseQueryConfigurationId) {
+        newAgentDatabaseQueryConfiguration =
+          await AgentDatabaseQueryConfiguration.create(
+            {
+              sId: generateModelSId(),
+              dataSourceWorkspaceId:
+                agentConfiguration.databaseQueryConfiguration
+                  .dataSourceWorkspaceId,
+              dataSourceId:
+                agentConfiguration.databaseQueryConfiguration.dataSourceId,
+              databaseId:
+                agentConfiguration.databaseQueryConfiguration.databaseId,
+            },
+            { transaction: t }
+          );
+      }
+
+      const newAgentSid = generateModelSId();
+      const newAgentConfiguration = await AgentConfiguration.create(
```

Comment:
here and for all the methods above, I think it's better to try and reuse the `create` functions so we avoid logic duplication

Reply:
Maybe you tried and this wasn't possible?